GCTC-NTGC/gc-digital-talent
# Updates Hydrogen to 2.0.5

## ✨ Summary

This pull request updates Hydrogen through [`2.0.4`](https://github.com/hydrogen-css/hydrogen/releases/tag/v2.0.4) to [`2.0.5`](https://github.com/hydrogen-css/hydrogen/releases/tag/v2.0.5) and includes the following changes:
* Migration from PostCSS/CSSnano/Autoprefixer to Lightning CSS
* A reduction (half!) in compile time
* The ability to configure `font_weight` values in `hydrogen.config.json`
* More accurate CSS selector validation
* A fix for old references to `clean.js`
* A fix for broken `generate_logs` configuration settings
* Fixes to various invalid CSS declarations in the app that existed prior to the update

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Grab the branch
2. Rerun `pnpm`'s installation (you might need to delete your `node_modules`)
3. Ensure you've got `2.0.5` installed
4. Run `pnpm build` to confirm the build compiles successfully
5. Check for visual diffs specific to the fixed bugged CSS declarations
